Added Annotation config.kubernetes.io/depends-on (#42313)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### config.kubernetes.io/depends-on

Type: Annotation

> Example: `config.kubernetes.io/depends-on: "OBJECT_REFERENCE"` 

Reviewer: This doesn't look right - is that a valid actual example?

Author: @sftim should I mention this in the example value?
```
annotations:
  config.kubernetes.io/depends-on: |
    /namespaces/default/Pod/pod-a,
    /namespaces/default/Pod/pod-c

```

Used on: All objects

This annotation is used on the target object to be modified to configure explicit dependency ordering. 
The annotation helps to determine the apply order, and dependency reconciliation while applying and deletion and prevents accidental premature deletion.

For eg. If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.

> See [Depends-on](https://kpt.dev/reference/annotations/depends-on/) for more information. 

Reviewer: ```suggestion
For more details, see [depends on](https://kpt.dev/reference/annotations/depends-on/) in the KPT
documentation.
```

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### config.kubernetes.io/depends-on

Type: Annotation

Example: `config.kubernetes.io/depends-on: "OBJECT_REFERENCE"` 

Used on: All objects

This annotation is used on the target object to be modified to configure explicit dependency ordering. 

> 

Reviewer 2: ```suggestion
```

Reviewer 2: Let's remove redundant spaces and lines.

> The annotation value accepts a list of resource references, delimited by commas.

Reviewer 2: ``` suggestion
This annotation is used on the target object to be modified to configure explicit dependency ordering. 
The annotation helps to determine the apply order, dependency reconciliation while applying and deletion and prevents accidental premature deletion.
See [Depends-on](https://kpt.dev/reference/annotations/depends-on/) for more information. 
The annotation value accepts a list of resource references, delimited by commas.
```

> **Resource reference**
> - **Group**: This represents the API group of the resource (e.g., apps, v1, extensions).
> - **Kind**: This indicates the type of resource (e.g., Pod, Service, Deployment).
> - **Name**: The name is the unique identifier of the specific resource instance.

Reviewer 2: ```suggestion
For eg. If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.
```
I think the link will explain what are group, kind, and name. Also, they weren't mentioned in the example and seem like popped up out of nowhere.

Reviewer 2: Please be noted the eg is already available on the linked page. However, we can add it in place of the Example section. If it seems repetitive I think the present Example section works fine as well, but less informative. Just feel like the explanation helps understanding annotation better.
WDYT, @sftim ?  need a second opinion.

Author: Hey @Ritikaa96, I have made the changes could you please give it a look

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### config.kubernetes.io/depends-on

Type: Annotation

> Example: `config.kubernetes.io/depends-on:`
> 
> ```yaml
>   annotations:
>     config.kubernetes.io/depends-on: "/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"
> ```

Reviewer 3: ```suggestion
Example: `config.kubernetes.io/depends-on:"/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"`
```

Used on: All objects

This annotation is used on the target object to be modified to configure explicit 
dependency ordering. The annotation helps to determine the apply order, and dependency 
reconciliation while applying and deletion and prevents accidental premature deletion.

> For example If the annotation `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` is added to `pod-b`,  it identifies that `pod`-a must be successfully applied prior to `pod-b` actuation.

Reviewer 3: ```suggestion
For example If the annotation `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a`
is added to `pod-b`, it identifies that `pod-a` must be successfully applied
prior to `pod-b` actuation.
```

Reviewer 3: We'd better document which tool is attaching this annotation.
A user needs to know when this appears and how to interpret it.

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

### config.kubernetes.io/depends-on

Type: Annotation

Example: `config.kubernetes.io/depends-on:`

```yaml
  annotations:
  config.kubernetes.io/depends-on: |
    /namespaces/default/Pod/pod-a,
    /namespaces/default/Pod/pod-c

> ```

Reviewer 3: This example is not correct.
The value of an annotation has to be a string.

Used on: All objects

This annotation is used on the target object to be modified to configure explicit dependency ordering. 

> The annotation helps to determine the apply order, and dependency reconciliation while applying and deletion and prevents accidental premature deletion.

Reviewer 3: please wrap the long lines as we do for other paragraphs.

> For example If `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a` annotation is added to `pod-b`,  it identifies that pod-a must be successfully applied prior to pod-a actuation.

Reviewer 3: Please be careful with capitals, spaces, and typos.

```suggestion
For example, if the annotation `config.kubernetes.io/depends-on: /namespaces/default/Pod/pod-a`
is added to a Pod `pod-b`, it identifies that `pod-a` must be successfully applied
prior to `pod-b` actuation.
```

> For more details, see [depends on](https://kpt.dev/reference/annotations/depends-on/) in the KPT documentation. 

Reviewer 4: I know I also first found this from KPT, but I wonder if a better permalink is https://github.com/kubernetes-sigs/cli-utils/blob/0b156cb0425fdb29a436d13f840a39039558c10e/README.md#explicit-dependency-ordering

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

depends-on
### config.kubernetes.io/depends-on

Type: Annotation

> Example: `config.kubernetes.io/depends-on:"/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"`

Reviewer: ```suggestion
Example: `config.kubernetes.io/depends-on: "/namespaces/default/Pod/pod-a,/namespaces/default/Pod/pod-c"`
```

which is used by Kustomize and similar third-party tools.
For example, Kustomize removes objects with this annotation from its final build output.

> depends-on

Reviewer 3: what's this?

Reviewer: ```suggestion
```
